ansible/roles/kafka/tasks/deploy.yml

```diff
@@ -15,7 +15,7 @@
       - "{{ zookeeper.port }}:2181"
 
 - name: wait until the Zookeeper in this host is up and running
-  action: shell echo ruok | nc -w 3 -i 1 {{ inventory_hostname }} {{ zookeeper.port }}
+  action: shell (echo ruok; sleep 1) | nc {{ inventory_hostname }} {{ zookeeper.port }}
```

Comment:
Actually, to use pipe in python.subprocess, we need `shell=True`.
But `shell=True` options is strongly discouraged due to security reason.
https://docs.python.org/2/library/subprocess.html#frequently-used-arguments

Could you share the test environment detail?
How can I reproduce that issue?